src/EnergyPlus/HVACFourPipeBeam.cc

```diff
@@ -0,0 +1,1302 @@
+// C++ Headers
+#include <cmath>
+
+// ObjexxFCL Headers
+#include <ObjexxFCL/Array.functions.hh>
+#include <ObjexxFCL/Fmath.hh>
+
+// EnergyPlus Headers
+#include <AirTerminalUnit.hh>
+#include <HVACFourPipeBeam.hh>
+#include <BranchNodeConnections.hh>
+#include <DataContaminantBalance.hh>
+#include <DataDefineEquip.hh>
+#include <DataEnvironment.hh>
+#include <DataHVACGlobals.hh>
+#include <DataIPShortCuts.hh>
+#include <DataLoopNode.hh>
+#include <DataPlant.hh>
+#include <DataPrecisionGlobals.hh>
+#include <DataSizing.hh>
+#include <DataZoneEnergyDemands.hh>
+#include <DataZoneEquipment.hh>
+#include <FluidProperties.hh>
+#include <General.hh>
+#include <GeneralRoutines.hh>
+#include <InputProcessor.hh>
+#include <NodeInputManager.hh>
+#include <OutputProcessor.hh>
+#include <PlantUtilities.hh>
+#include <Psychrometrics.hh>
+#include <ReportSizingManager.hh>
+#include <ScheduleManager.hh>
+#include <UtilityRoutines.hh>
+#include <CurveManager.hh>
+
+
+namespace EnergyPlus {
+
+namespace FourPipeBeam {
+
+	Array1D< std::shared_ptr< HVACFourPipeBeam > > FourPipeBeams; // dimension to number of machines
```

Comment:
@EnergyArchmage Do you intend for these HVACFourPipeBeam objects to be used and referenced by other objects? Basically do these need to be std::shared_ptr or can they be std::unique_ptr? They very well could require the use of std::shared_ptr. 

For example, will an air terminal or any other object hold a `std::shared_ptr< HVACFourPipeBeam >` object within their class?

@Myoldmopar fyi